### ISSUE TITLE
Regression in error/build split?

### ISSUE THREAD

**Issue by edsko** [atom-haskell/ide-haskell-cabal#16]
I'm getting output such as

```
/path/to/project/cmdline/src/Foo/Bar/Client.hs: 66, 12
• No instance for (aeson-0.11.3.0:Data.Aeson.Types.Class.FromJSON
```

in my Errors tab, with the rest of the error message in the Build tab:

```
[ 9 of 12] Compiling Foo.Bar.Client ( src/Foo/Bar/Client.hs, /path/to/project/dist-newstyle/build/x86_64-osx/ghc-8.0.2/package-0.1.0.0/c/exe/build/exe/exe-tmp/Foo/Bar/Client.o )
                         NoContent)
        arising from a use of ‘client’
    • In the expression: client API.proxy
      In an equation for ‘baz’: baz = client API.proxy
```

**Comment by lierdakil** [comment]
I've pushed d3005dd, if you could test that, this would be great. I'm not too sure on reproduction conditions, but looking at the code I wrote when half-asleep, sure, it might, with some buffering peculiarities, end up leading to observed behavior.

**Comment by edsko** [comment]
Yup, I can confirm that that fixes it! :-)

**Comment by lierdakil** [comment]
Great, thanks. Published as v1.7.5.

**Comment by edsko** [comment]
Awesome :)
